solvers/ceres_utils.h

```diff
@@ -35,21 +35,28 @@ inline T NormalizeAngle(const T & angle_radians)
 /*****************************************************************************/
 /*****************************************************************************/
 
-class AngleLocalParameterization
-{
-public:
-  template<typename T>
-  bool operator()(
-    const T * theta_radians, const T * delta_theta_radians,
-    T * theta_radians_plus_delta) const
-  {
-    *theta_radians_plus_delta = NormalizeAngle(*theta_radians + *delta_theta_radians);
+// Defines a manifold for updating the angle to be constrained in [-pi to pi).
+class AngleManifold {
+ public:
+  template <typename T>
+  bool Plus(const T* x_radians,
+            const T* delta_radians,
+            T* x_plus_delta_radians) const {
+    *x_plus_delta_radians = NormalizeAngle(*x_radians + *delta_radians);
     return true;
   }
 
-  static ceres::LocalParameterization * Create()
-  {
-    return new ceres::AutoDiffLocalParameterization<AngleLocalParameterization, 1, 1>;
+  template <typename T>
+  bool Minus(const T* y_radians,
+             const T* x_radians,
+             T* y_minus_x_radians) const {
+    *y_minus_x_radians =
+        NormalizeAngle(*y_radians) - NormalizeAngle(*x_radians);
```

Comment:
```suggestion
        NormalizeAngle(*y_radians - *x_radians);
```

No? 2 subtracted angles are not necessarily normalized just because their initial angles were. How about making minus just call plus setting `x_radians` to `-x_radians`?

Reply:
You are right. Fixed.

> How about making minus just call plus setting x_radians to -x_radians?

This is inconvenient because of the pointer type used in the parameter.